sdk/storage/storage-blob/src/utils/utils.node.ts

```diff
@@ -96,11 +91,13 @@ export async function streamToBuffer2(
         chunk = Buffer.from(chunk, encoding);
       }
 
-      // How much data needed in this chunk
-      const chunkLength = pos + chunk.length > bufferSize ? bufferSize - pos : chunk.length;
+      if (pos + chunk.length > bufferSize) {
```

Comment:
This changes the behavior of this function. The original author @XiaoningLiu may want to take a look.

Reply:
I agree this is a behavior change, but it was intentional because I think the previous code was wrong.  The comment above the function says:

> @throws {RangeError} If buffer size is not big enough.

The previous code would throw if it read another chunk after the buffer was full, but it would **not** throw if the last chunk exceeded the buffer.  I think this was a copy/paste bug caused by copying the code from `streamToBuffer()`.

Reply:
I agree the new way is more consistent.